pkg/module_manager/models/hooks/global_hook.go

```diff
@@ -0,0 +1,122 @@
+package hooks
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	types2 "github.com/flant/addon-operator/pkg/hook/types"
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/pkg/module_manager/models/hooks/kind"
+	"github.com/flant/shell-operator/pkg/hook/types"
+)
+
+// GlobalHook is a representation of the hook, which not belongs to any module
+type GlobalHook struct {
+	executableHook
```

Comment:
Well, ideologically you are right. But executableHook it not a struct, it's an interface. So GlobalHook must behave as executableHook.
It's a good point, but otherwise we have to reimplement all `executableHook` methods like `GetName()`, `GetPath()` and so on.
<img width="732" alt="image" src="https://github.com/flant/addon-operator/assets/4477098/86eda7e8-e8a3-4e2b-b16a-c872edcef1af">

It's really looks like hugo pile of copy-paste. While it's an interface, I don't see any real problems

Reply:
It seems like copy pasting methods is more transparent than embedding, and twice more transparent than embedding and interface.

Reply:
Embedding interfaces is ok. It's a common pattern like `SocketConn` with interface `net.Conn`, `ReverseSort` and so on.